evap/evaluation/management/commands/anonymize.py

```diff
@@ -111,22 +129,100 @@ def replace_usernames_emails(random_usernames):
                 new_domain = Command.new_institution_domain if is_institution_domain else Command.new_external_domain
                 user.email = user.username + '@' + new_domain
 
-            user.save()
+            user.login_key = key
+            user.login_key_valid_until = valid_until
```

Comment:
if i understand correctly, you're giving every user a login key (plus valid until date). on production, only some of the users have login keys, and i would suggest to give only those users a new one. actually, i would use `ensure_valid_login_key` (so, first delete the existing key, and then call that method) for this, to not duplicate the code that generates the key.

additionally, in the past we had problems that the test data went stale, i.e. that after some months the states of the courses changed. to avoid the same happening here, i would suggest to set the validity to the year 2100 (and perhaps half of them to some past date to reflect that not all login keys are valid on production)

Reply:
In the current version, I use `refresh_login_key`, which automatically overwrites the last key.
Then, I change the valid until date by 100 years in the past or future.

Reply:
Jumping in here since I just changed some of the code that does the login key generation:
`refresh_login_key` was just renamed to `reset_login_key_validity` and does exactly that - it resets the validity, but does not touch the actual key. The old name was misleading. You will need `ensure_valid_login_key` as @karyon proposed. Also, you will need to delete the old key before the call as the method keeps a key if it is still valid.

Reply:
The intention was to provide diverse test data with
* some users with valid keys,
* some users with invalid keys and
* some users without any key at all.
So should we just provide data for the first case?

Reply:
Okay, fixed that.